.github/workflows/my-ci-pipeline.yml

```
@@ -7,12 +7,28 @@ on:
     paths-ignore:
       - 'READ.md'
 
-
 jobs:
   build:
     name: Build and test
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
     - run: docker build --tag todo-app:test --target test .
-    - run: docker run todo-app:test
\ No newline at end of file
+    - run: docker run todo-app:test
+
+  deploy:
+    name: Push to Docker Hub and Trigger App Service Deploy
+    runs-on: ubuntu-latest
+    needs: build
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/exercise=9' }}
+    steps:
+    - uses: actions/checkout@v2
```

I know this is because it matches the instructions, but worth noting that you're getting some warnings in the pipeline because we're a bit out of date - should be v4!
```suggestion
    - uses: actions/checkout@v4
```

It's the warnings on your pipeline run pages [like this one](https://github.com/SidrahJ77/DevOps-Course-Starter/actions/runs/9680473874) that look like:
> Node.js 16 actions are deprecated. ...